test/framework/config.py

```diff
@@ -570,6 +572,89 @@ def test_strict(self):
+        regex = re.compile(os.path.join(tmpdir, r'easybuild-foo-1\.2\.3-19700101\.094651\.log\.1$'))
+        self.assertTrue(regex.match(res), "Pattern '%s' matches '%s'" % (regex.pattern, res))
+
+        # adding salt ensures a unique filename (pretty much)
+        prev_log_filenames = []
+        for i in range(10):
+            res = get_log_filename('foo', '1.2.3', date='19700101', timestamp='094651', add_salt=True)
+            regex = re.compile(os.path.join(tmpdir, r'easybuild-foo-1\.2\.3-19700101\.094651\.[a-zA-Z]{5}\.log$'))
+            self.assertTrue(regex.match(res), "Pattern '%s' matches '%s'" % (regex.pattern, res))
+            self.assertTrue(res not in prev_log_filenames)
+            prev_log_filenames.append(res)
+
+    def test_log_file_format(self):
+        """Test for log_file_format()."""
+
+        # first test defaults -> no templating when no values are provided
+        self.assertEqual(log_file_format(), 'easybuild-%(name)s-%(version)s-%(date)s.%(time)s.log')
+        self.assertEqual(log_file_format(return_directory=True), 'easybuild')
+
+        # test whether provided values are used to complete template
+        ec = {'name': 'foo', 'version': '1.2.3'}
+        res = log_file_format(ec=ec, date='20190322', timestamp='094356')
+        self.assertEqual(res, 'easybuild-foo-1.2.3-20190322.094356.log')
+
+        res = log_file_format(return_directory=True, ec=ec, date='20190322', timestamp='094356')
+        self.assertEqual(res, 'easybuild')
+
+        # partial templating is done when only some values are provided...
+        self.assertEqual(log_file_format(ec=ec), 'easybuild-foo-1.2.3-%(date)s.%(time)s.log')
+        res = log_file_format(date='20190322', timestamp='094356')
+        self.assertEqual(res, 'easybuild-%(name)s-%(version)s-20190322.094356.log')
+
+        # also try with a custom setting
+        init_config(args=['--logfile-format=eb-%(name)s-%(date)s,log-%(version)s-%(date)s-%(time)s.out'])
+        self.assertEqual(log_file_format(), 'log-%(version)s-%(date)s-%(time)s.out')
+        self.assertEqual(log_file_format(return_directory=True), 'eb-%(name)s-%(date)s')
+
+        res = log_file_format(ec=ec, date='20190322', timestamp='094356')
+        self.assertEqual(res, 'log-1.2.3-20190322-094356.out')
+
+        res = log_file_format(return_directory=True, ec=ec, date='20190322', timestamp='094356')
+        self.assertEqual(res, 'eb-foo-20190322')
+
+    def test_log_path(self):
+        """Test for log_path()."""
+        # default
+        self.assertEqual(log_path(), 'easybuild')
+
+        # providing template values doesn't affect the default
+        ec = {'name': 'foo', 'version': '1.2.3'}
+        res = log_path(ec=ec)
+        self.assertEqual(res, 'easybuild')
+
+        # reconfigure with value for log directory that includes templates
+        init_config(args=['--logfile-format=easybuild-%(name)s-%(version)s-%(date)s-%(time)s,log,txt'])
```

Comment:
I don't think this option should be valid, isn't your log file going to be `log,txt` in this case?

Reply:
Nope, just `log`, everything from the last comma is ignored

Reply:
Oops, yeah. It'll probably be `log` rather than `log,txt`, but you're right, this doesn't make sense...

Reply:
fixed in 884413b1c